File: order/src/main/java/com/example/order/module/domain/order/enitity/Order.java
package com.example.order.module.domain.order.enitity;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;
import java.time.LocalDate;

@Entity
@Setter @Getter @EqualsAndHashCode(of ="id")
@AllArgsConstructor @NoArgsConstructor
@Table(name = "tb_order")
public class Order {

    @Id @GeneratedValue
    private Long id;

    private Long productId;

    private Long colorId;

    private Long userId;

    private int amount;

    private String address;

    private int payment;

    private LocalDate createAt;

    private LocalDate updateAt;

}

File: customer-view/shared/interface/order/Order.ts
interface Order{
    products : RegisterOrder[]
    address : string | any
    phone : string | any
}

interface RegisterOrder{
    productId : number,
    colorId : number,
    sizeId : number | null,
    amount : number,
    payment : number,
    couponId : number | null,
    productName: string
}